ipn/ipnlocal/web_client.go

```diff
@@ -88,3 +94,38 @@ func (b *LocalBackend) handleWebClientConn(c net.Conn) error {
 	s := http.Server{Handler: b.webClient.server}
 	return s.Serve(netutil.NewOneConnListener(c, nil))
 }
+
+func (b *LocalBackend) updateWebClientTCPPortNetMapAddrListenersLocked() {
+	if b.netMap == nil {
+		return
+	}
+
+	addrs := b.netMap.GetAddresses()
+	for i := range addrs.LenIter() {
+		addrPort := netip.AddrPortFrom(addrs.At(i).Addr(), webClientPort)
+		if _, ok := b.webClientListeners[addrPort]; ok {
+			continue // already listening
+		}
```

Comment:
you don't need to shut down old running listeners in this list too if they're no longer in addrs?

Reply:
we don't need to do quite the same thing serve is doing because you can't add or remove ports for the web client like you can with serve.  I did however update `b.WebClientShutdown` to close the listeners.